### ISSUE TITLE
[Due for payment 2025-10-17] [$250] Cannot make an expense negative

### ISSUE THREAD

**Issue by m-natarajan** [Expensify/App#53441]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.70-1
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught on HybridApp, is this reproducible on New Expensify Standalone?:**
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** @twisterdotcom 
**Slack conversation** (hyperlinked to channel name): [ts_external_expensify_bugs](https://expensify.slack.com/archives/C049HHMV9SM/p1733220192826509)

## Action Performed:

1. SmartScan a receipt
2. Open the expense
3. Open the amount
4. Try to correct it to be negative. No way to add brackets or - sign.

## Expected Result:
User able to enter minus, plus or Brackets
## Actual Result:
User unable to enter minus, plus or Brackets
## Workaround:
Unknown
## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Standalone
- [ ] Android: HybridApp
- [ ] Android: mWeb Chrome
- [ ] iOS: Standalone
- [ ] iOS: HybridApp
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

<details>
  <summary>Add any screenshot/video evidence</summary>

  
</details>

![Shottr 2024-12-03 10 02 13](https://github.com/user-attachments/assets/55854062-5641-4cbf-852d-e2632aa7b12c)

https://github.com/user-attachments/assets/b8174146-d638-4e1b-8177-c6c0fa69d8db

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021866173019746429432</li>
        <li>Upwork Job ID: 1866173019746429432</li>
        <li>Last Price Increase: 2025-01-11</li>
<li>Automatic offers: </li>
<ul>
<li>ZhenjaHorbach | Reviewer | 105292850</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @isabelastisser</details>

**Comment by garrettmknight** [comment]
Potentially a new feature - asking in thread.

**Comment by pasyukevich** [comment]
Hi, I'm Eugene from Callstack - expert contributor group. I’d like to work on this issue.

**Comment by pasyukevich** [comment]
Should I create just a PR here without the proposal?

**Comment by pasyukevich** [comment]
In this case, should we allow the entry of a plus sign, calculate the combinations result, and send the computed result to the API?

**Comment by pasyukevich** [comment]
@ZhenjaHorbach 

Yes, I think this would be great - we need some discussion here for the expected behavior

**Comment by marcaaron** [comment]
>User able to enter minus, plus or Brackets

@twisterdotcom is there more context on why a user would use these characters when editing an expense? I think some more information on what we're trying to achieve might help contributors understand what we're trying to do here and help move this along. Thanks!

**Comment by marcaaron** [comment]
Maybe we can try to track down someone who has more familiarity with negative expenses to help investigate this. I'll be going on leave sometime in the next few weeks and probably won't get to this before then.

**Comment by twisterdotcom** [comment]
>more context on why a user would use these characters when editing an expense?

Sure, I see three fairly likely use cases:

1. A user is tracking personal expenses and chooses to invert everything SmartScan does to make expenses negative offsetting a positive single expenses total they create. (Uncommon, but it does happen on oldDot).
2. Similar but a business user will do the same for pre-approval cash amount, and add SmartScanned expenses to offset that. (Again uncommon, but more common than 1)
3. Like me, users may be indebted to the company a small amount, and this is how they are able to pay it back (by reneging on future expense reimbursements).
4. SmartScan scans the amount wrong. (The most common)

**Comment by nkuoch** [comment]
Yes, let's allow typing the minus sign. Let me know if sending it to the api doesn't act as expected and I can look into it.

**Comment by pasyukevich** [comment]
Great! 

I will prepare the update then

**Comment by twisterdotcom** [comment]
How is this going @pasyukevich?

**Comment by pasyukevich** [comment]
Coming back here. Working on it

**Comment by pasyukevich** [comment]
@nkuoch What should we see in the report after this update?  (current amount  and changes history)

I don't see any errors connected with the updated amount

requestID - 8ff5223a4cfcb17f-WAW 

<img width="652" alt="image" src="https://github.com/user-attachments/assets/b6cf1cdb-d424-4557-b40a-64c7f9114e2e" />

**Comment by nkuoch** [comment]
displayed amount should show `($75.29)`. When clicking on it edit field should show `-75.29`

**Comment by nkuoch** [comment]
hold on, we're discussing internally whether we should use brackets or minus sign

**Comment by pasyukevich** [comment]
sure, I will put this on pause

**Comment by johncschuster** [comment]
I've updated the title to `[Hold for discussion]` while the design chat is ongoing. Will remove the hold once we've dialed it in!

**Comment by pasyukevich** [comment]
Is it ready to work on?
As I see from the thread the discussion is finished.

**Comment by pasyukevich** [comment]
Updates: 

Working on it

**Comment by dannymcclain** [comment]
cc @Expensify/design ☝

**Comment by shawnborton** [comment]
Lovely!

**Comment by pasyukevich** [comment]
Updates:

- Working on an update

**Comment by pasyukevich** [comment]
Can we update the ticket description? It seems more like a feature update for the ‘Create Manual Expense’  and 'Edit Amount' than an issue.

Should I also include the update for other places where we need to render this amount in negative too?

**Comment by pasyukevich** [comment]
> Can we update the ticket description? It seems more like a feature update for the ‘Create Manual Expense’ and 'Edit Amount' than an issue.
> 
> Should I also include the update for other places where we need to render this amount in negative too?

Bump here

**Comment by pasyukevich** [comment]
After this update should we have any places where negative value should not be shown?

**Comment by pasyukevich** [comment]
Updates: 

 - UI work done
 - BE request and response with negative  - handles properly 
 - Now focused on final checks

**Comment by pasyukevich** [comment]
@nkuoch 

I've noticed that when we send a negative value with `UpdateMoneyRequestAmountAndCurrency` the API responds with absolute `modifiedAmount` 

requestID: `90ab96bc8a8aee49-SJC`

**Comment by pasyukevich** [comment]
@dannymcclain @shawnborton 

What do you think about the ‘minus’ sign appearing between the currency symbol and the amount?

Since this is an input field, we need to modify the input handlers to ensure the minus sign is reflected in the value but not shown in the input field itself. It should still be positioned correctly on the left.

From the user’s perspective on a desktop, they will type the minus sign in one place (via the keyboard), but it will appear in a different position in the field.

I’m attaching two screenshots: one with the keyboard visible and one without, to illustrate the current behavior.

<img width="411" alt="Image" src="https://github.com/user-attachments/assets/17eb8be9-e905-42d3-b71a-5160affdf3c8" />

<img width="374" alt="Image" src="https://github.com/user-attachments/assets/d1c45216-501d-4587-90e9-1f4e454665c6" />

**Comment by dannymcclain** [comment]
cc @Expensify/design for a gut check here, but this is my thinking:

>What do you think about the ‘minus’ sign appearing between the currency symbol and the amount?

I think this looks really weird and doesn't really feel like how I typically see negative money amounts.

>Since this is an input field, we need to modify the input handlers to ensure the minus sign is reflected in the value but not shown in the input field itself. It should still be positioned correctly on the left.
>
>From the user’s perspective on a desktop, they will type the minus sign in one place (via the keyboard), but it will appear in a different position in the field.

I think the idea here is that we won't actually allows users to _type_ the minus sign, but rather they will need to use the `Flip` button to change between positive and negative amounts. Does that make it easier to position the minus symbol to the left of the currency symbol? i.e. `-$4.00`

**Comment by dubielzyk-expensify** [comment]
> I think this looks really weird and doesn't really feel like how I typically see negative money amounts.

Agree. I think the negative symbol should appear at the start always. There might be currencies I've not accounted for, but I believe this'll be more obvious

**Comment by shawnborton** [comment]
Yup, totally agree with that.

**Comment by pasyukevich** [comment]
Got it

Thanks for the clarification, I will proceed then

**Comment by pasyukevich** [comment]
Updates: 

 - Keep working on the update - gluing with API submit flow

**Comment by pasyukevich** [comment]
> @nkuoch
> 
> I've noticed that when we send a negative value with UpdateMoneyRequestAmountAndCurrency the API responds with absolute modifiedAmount
> 
> requestID: 90ab96bc8a8aee49-SJC

@nkuoch Could you take a look?

**Comment by pasyukevich** [comment]
Updates: 
 
- Removing redundant code from the previous implementation, recording videos to open the PR

**Comment by nkuoch** [comment]
> @nkuoch Could you take a look?

Asking in slack https://expensify.slack.com/archives/C049HHMV9SM/p1738936366214349?thread_ts=1733220192.826509&cid=C049HHMV9SM

**Comment by pasyukevich** [comment]
PR is ready for review

I've prepared steps only with  the creation of the expense with minus since edit submission does not apply it on the API side

https://github.com/Expensify/App/pull/56092

**Comment by pasyukevich** [comment]
> @nkuoch
> 
> I've noticed that when we send a negative value with UpdateMoneyRequestAmountAndCurrency the API responds with absolute modifiedAmount
> 
> requestID: 90ab96bc8a8aee49-SJC

@nkuoch I've also noticed that it is not saving a negative amount for new expenses for already existent reports 

`"requestID": "91790da539dbee3a-LAX",`

and for workspaces, it saves negative values only for the  messages - `"requestID": "917913451f66ee44-WAW",`

**Comment by mallenexpensify** [comment]
[Per diem - LHN preview is missing negative sign when the amount is negative](https://github.com/Expensify/App/issues/57308#top)
👋 Should I put the above on hold pending this issue?  I'm unsure how we're handling negative expenses, didn't see any mention of them [in testrai](https://expensify.testrail.io/index.php?/new_search/results&queries[]=negative&category[]=2)l.

**Comment by pasyukevich** [comment]
> > @nkuoch
> > 
> > I've noticed that when we send a negative value with UpdateMoneyRequestAmountAndCurrency the API responds with absolute modifiedAmount
> > 
> > requestID: 90ab96bc8a8aee49-SJC
> 
> @nkuoch I've also noticed that it is not saving a negative amount for new expenses for already existent reports
> 
> "requestID": "91790da539dbee3a-LAX",
> 
> and for workspaces, it saves negative values only for the messages - "requestID": "917913451f66ee44-WAW",

@nkuoch Also linking here the API problem with the negative split. - https://github.com/Expensify/App/pull/56092#issuecomment-2701439021

**Comment by nkuoch** [comment]
Sorry, not a priority right now.
Also, what's been said on [slack](https://expensify.slack.com/archives/C049HHMV9SM/p1739480341384079?thread_ts=1733220192.826509&cid=C049HHMV9SM) to avoid IOU issues:

- If you create a negative expense then we could filter the next page to workspaces only.
- Let’s not allow negatives in the DM create menu unless it’s a workspace chat either

> I've noticed that when we send a negative value with UpdateMoneyRequestAmountAndCurrency the API responds with absolute modifiedAmount

Did you call it from an IOU report?

**Comment by pasyukevich** [comment]
Got it, thanks 

I’ve tried with Workspace and also faced issues:     
- With `UpdateMoneyRequestAmountAndCurrency`  response sending back positive modified amount in case of negative input ("requestID":` "91cba9ec59a2eeb5-SJC"`,)  and negative if it was positive input ( `"requestID": "91cbdad04a4d027a-WAW"`,)
- WIth negative split bill in workspace also faced with an error - `"requestID": "91cbdf30ce93c3ec-WAW"`
- With `RequestMoney` I’ve also faced that in Onyx coming update with negative  if it was submitted positive - same value as was submitted saved only for messages  (` "requestID": "91cbe30e7fdcb813-WAW”`)

**Comment by pasyukevich** [comment]
> Got it, thanks
> 
> I’ve tried with Workspace and also faced issues:   
> 
> * With `UpdateMoneyRequestAmountAndCurrency`  response sending back positive modified amount in case of negative input ("requestID":` "91cba9ec59a2eeb5-SJC"`,)  and negative if it was positive input ( `"requestID": "91cbdad04a4d027a-WAW"`,)
> * WIth negative split bill in workspace also faced with an error - `"requestID": "91cbdf30ce93c3ec-WAW"`
> * With `RequestMoney` I’ve also faced that in Onyx coming update with negative  if it was submitted positive - same value as was submitted saved only for messages  (` "requestID": "91cbe30e7fdcb813-WAW”`)

@nkuoch Kindly bump

**Comment by pasyukevich** [comment]
@nkuoch Kindly bump

**Comment by nkuoch** [comment]
Sorry for the delay

> With UpdateMoneyRequestAmountAndCurrency response sending back positive modified amount in case of negative input

Yes it's normal that in `transactions_6427255952362233852`, amounts are with the opposite sign, as it's how it's stored in our transactions table in db. Which is why we then do things like this everywhere in the code:

<img width="990" alt="Image" src="https://github.com/user-attachments/assets/a6a1787c-e39c-4a01-a738-450c1a9494a1" />

**Comment by nkuoch** [comment]
> WIth negative split bill in workspace also faced with an error

CreateIOUSplit doesn't currently support negative expenses. Asking in [slack](https://expensify.slack.com/archives/C049HHMV9SM/p1742910816688119?thread_ts=1733220192.826509&cid=C049HHMV9SM) if we should.

**Comment by pasyukevich** [comment]
Thanks, got it. looks like, in such case, I need to invert the logic on FE side for displaying and send

**Comment by pasyukevich** [comment]
![Image](https://github.com/user-attachments/assets/f72b3c4b-b483-4115-ad28-ee1758073c2d)

@nkuoch Faced with an issue on the API side when editing a negative amount to a positive one   (the old amount in the message becomes positive)

`requestID: "94d056db8919ee3a-SJC"`

**Comment by nkuoch** [comment]
Isn't is just a matter of:

>  I need to invert the logic on FE side for displaying and send

?

Where are you testing? On a branch? Doing what action?

**Comment by pasyukevich** [comment]
Oh, looks like you are right, thanks

I'm testing on dev, this branch - https://github.com/Expensify/App/pull/56092 

Action - edit for already created expense

**Comment by pasyukevich** [comment]
New draft PR started to address the issues - https://github.com/Expensify/App/pull/65001

**Comment by johncschuster** [comment]
# BugZero Checklist:

- [ ] **[Contributor]** Classify the bug:

<details>
<summary>Bug classification</summary>


Source of bug:
  - [ ] 1a. Result of the original design (eg. a case wasn't considered)
  - [ ] 1b. Mistake during implementation
  - [ ] 1c. Backend bug
  - [ ] 1z. Other:

Where bug was reported:
  - [ ] 2a. Reported on production (eg. bug slipped through the normal regression and PR testing process on staging)
  - [ ] 2b. Reported on staging (eg. found during regression or PR testing)
  - [ ] 2d. Reported on a PR
  - [ ] 2z. Other:

Who reported the bug:
  - [ ] 3a. Expensify user
  - [ ] 3b. Expensify employee
  - [ ] 3c. Contributor
  - [ ] 3d. QA
  - [ ] 3z. Other:

</details>

- [ ] **[Contributor]** The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake.

    Link to comment:

- [ ] **[Contributor]** If the regression was CRITICAL (e.g. interrupts a core flow) A discussion in [#expensify-open-source](https://app.slack.com/client/E047TPA624F/C01GTK53T8Q) has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner.

    Link to discussion:

- [ ] **[Contributor]** If it was decided to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps using the template below to ensure the same bug will not reach production again.

<details>
<summary>Regression Test Proposal Template</summary>


- [ ] **[BugZero Assignee]** Create a GH issue for creating/updating the regression test once above steps have been agreed upon.

    Link to issue:

## Regression Test Proposal
### Precondition:


-

### Test:


1.

Do we agree 👍 or 👎


</details>

**Comment by ZhenjaHorbach** [comment]
> ### Payment Summary (to be issued after regression window has passed)
> Contributor: [@pasyukevich](https://github.com/pasyukevich) does not require payment (Contractor) Contributor+: [@ZhenjaHorbach](https://github.com/ZhenjaHorbach) paid $250 via [Upwork](https://www.upwork.com/nx/wm/offer/105292850)
> 
> [@ZhenjaHorbach](https://github.com/ZhenjaHorbach) heads up! I will not be online the day this issue requires payment (Friday, July 4th). If you require payment on that day, please feel free to bump anyone from the BZ Team in #expensify-open-source, otherwise, I will issue payment when I return on Monday, July 7th.

We decided to revert a PR to fix some issues 
So let's remove Due for payment !

**Comment by pasyukevich** [comment]
@nkuoch Looks like this issue can be fixed only on the API side 

https://github.com/Expensify/App/issues/64863

**Comment by trjExpensify** [comment]
@pasyukevich @ZhenjaHorbach what are the next steps here, and do we have an ETA to get a new PR deployed? We're considering including this in a product update email by mid-month, so we'd ideally be looking to be testing it next week to feel confident in shipping the comms about it. Thanks!

**Comment by pasyukevich** [comment]
@trjExpensify I'm working on addressing issues in the draft PR 
Next step - to address all of them (draft pr with linked issues - https://github.com/Expensify/App/pull/65001

And we have one issue that looks like API related (https://github.com/Expensify/App/issues/64863)

**Comment by pasyukevich** [comment]
Regarding the exact ETA, it is hard to say right now 
I will post it after fixing more issues

**Comment by trjExpensify** [comment]
>And we have one issue that looks like API related (https://github.com/Expensify/App/issues/64863)

Is @nkuoch looking into that if it's BE?

**Comment by nkuoch** [comment]
For the [back end](https://github.com/Expensify/App/issues/64863) issue I can fix the report preview that we're using in the email with these PRs:
- https://github.com/Expensify/Web-Expensify/pull/47591
- https://github.com/Expensify/Auth/pull/16092

Which means that if I submit a negative report like this:

<img width="1724" height="902" alt="Image" src="https://github.com/user-attachments/assets/92f8e745-1d75-4a22-9d52-04084ed33f0c" />

Then the preview title would be like this:

<img width="1362" height="787" alt="Image" src="https://github.com/user-attachments/assets/1cc228b8-5b74-4737-bf95-7e417a251b4b" />

Does it make sense to write that someone **owes** a negative amount?

**Comment by shawnborton** [comment]
> is there an issue for that?

I don't believe so, I can make one.

**Comment by johncschuster** [comment]
Just waiting on the BE changes before issuing payment.

**Comment by pasyukevich** [comment]
@nkuoch could you also take a look at these issues?  They also look for me related to the API side 

- https://github.com/Expensify/App/issues/64868 

Reproduced it, saw that positive is always coming from the API in this case 

- https://github.com/Expensify/App/issues/64876 

Notification

**Comment by pasyukevich** [comment]
@johncschuster FE work is also ongoing with fixes - https://github.com/Expensify/App/pull/65001

**Comment by johncschuster** [comment]
Thanks! In that case, I'll remove the `Due for payment` status in the issue title until this is ready.

**Comment by nkuoch** [comment]
Yeah, I'll let you know once https://github.com/Expensify/Auth/pull/16092 is deployed so you can retest

**Comment by JmillsExpensify** [comment]
Any chance we think we'll get this done by next week? Asking because if not, then we need to remove it from the August product marketing.

**Comment by pasyukevich** [comment]
> I have a question regarding this issue - [#64872](https://github.com/Expensify/App/issues/64872)
> 
> Should we show a pay button for the negative amount or smth different?

kindly bump

**Comment by johncschuster** [comment]
Not overdue. Progress has been made, we're just waiting on a deploy and an answer to an outstanding question.

**Comment by pasyukevich** [comment]
> Hi there,
> 
> I will be on vacation from this Friday, August 15, until August 30. Should we assign someone else to handle this in my absence? Once the API is merged, I’ll be happy to finalize it after I’m back.
> 
> Thanks!

Just following up — should we assign somebody to these issues during my absence so the work can continue, or would it be fine to resume once I’m back?

**Comment by VickyStash** [comment]
@pac-guerreiro is going to take it over while Eugene is OOO!

**Comment by johncschuster** [comment]
Thanks, @VickyStash! @pac-guerreiro, I've assigned the issue to you!

**Comment by waterim** [comment]
Hello! Im Artem from Callstack and would like to take over this issue as @pac-guerreiro is busy right now with another critical issue!

I've started investigating a bit today and on friday and will post my udpates soon

**Comment by pasyukevich** [comment]
Latest update was posted by @waterim here - https://github.com/Expensify/App/pull/65001#issuecomment-3243283281 

I've come back to this issue and continue working here 

Now, I'm validating all issues one by one to address what is  left

**Comment by pasyukevich** [comment]
Updates for today: 

   **UI/Layout**

    - Fixed Flip button UI 
    - Changed currency dropdown → button with selected currency (as in original PR)
    -  Restored layout to match original PR

 **Issues Reproduced during the verification** 
     -	Negative expenses submission to DM → reproduced
   
  
 **Reported Issues verification status:**

$[#65147](https://github.com/Expensify/App/issues/65147) - not reproduced 
$[#64851](https://github.com/Expensify/App/issues/64851) - not reproduced 
$[#64853](https://github.com/Expensify/App/issues/64853) - **reproduced** 
$[#64855](https://github.com/Expensify/App/issues/64855) - not reproduced 
$[#64856](https://github.com/Expensify/App/issues/64856) - not reproduced
$[#64857](https://github.com/Expensify/App/issues/64857) - not reproduced
$[#64863](https://github.com/Expensify/App/issues/64863) - not reproduced
$[#64868](https://github.com/Expensify/App/issues/64868) - not reproduced
$[#64870](https://github.com/Expensify/App/issues/64870) - not reproduced
$[#64872](https://github.com/Expensify/App/issues/64872)  - (status pending / unclear)
$[#64874](https://github.com/Expensify/App/issues/64874) - **reproduced** 
$[#64876](https://github.com/Expensify/App/issues/64876) - (status pending / unclear)
$[#64880](https://github.com/Expensify/App/issues/64880) - (status pending / unclear)
$[#64896](https://github.com/Expensify/App/issues/64896) - (status pending / unclear)
$[#64927](https://github.com/Expensify/App/issues/64927) - (status pending / unclear)
$[#64964](https://github.com/Expensify/App/issues/64964) - (status pending / unclear)


Next steps here: 
  1. Finish issue validation from the list 
  2. Fix/identify API 
  3. Fix the Negative expenses submission to the DM issue
  4. Open the PR

**Comment by ZhenjaHorbach** [comment]
> Thanks!
> 
> I meant having a round of regression tests for this functionality on the local build once I fix all the issues in the PR.
> 
> Would it be possible to have QAs additionally validate this feature?
> 
> Currently, it hasn’t been deployed to any environment.

I suppose we can trigger adhoc builds for the PR 
And then share them with the QA team

**Comment by pasyukevich** [comment]
Currently, I'm working on the reproduced issues that are related to the FE side 

> 1. Negative expenses submission to DM → reproduced
> 2. $https://github.com/Expensify/App/issues/64853 - reproduced
> 3. $https://github.com/Expensify/App/issues/64872 - reproduced
> 4. $https://github.com/Expensify/App/issues/64880 - reproduced
> 5. $https://github.com/Expensify/App/issues/64964 - reproduced


Yes, I think that the best order is next: 

1. Fix issues (BE and FE in parallel) 
2. Ask QA for a regression test of the adhoc frontend PR
3. Fix what was found 
4. Open PR

**Comment by pasyukevich** [comment]
Updates: 

 - Tests are passing now 
 - Rechecked all issues one more time 
 - Fixed what I found 

Now: 

 - Doing final checks on mobile

**Comment by pasyukevich** [comment]
@johncschuster

Tapping you in — the local PR is ready for a regression test round

**Comment by pasyukevich** [comment]
> > [@flaviadefaria](https://github.com/flaviadefaria)
> > I've done all my checks
> > Current PR link - [#65001](https://github.com/Expensify/App/pull/65001)
> > Next steps for me:
> > 
> > * Update test steps
> > * Record new videos
> > 
> > Next steps - regression tests on local builds
> 
> I suppose that after the test steps are updated, we can share the PR with QA

I think we can do it already

**Comment by ZhenjaHorbach** [comment]
Asked [here](https://expensify.slack.com/archives/C08CZDJFJ77/p1758099675786289) to run builds !

**Comment by pasyukevich** [comment]
@IuliiaHerets 
Looks like you are on stage, this is not merged yet 

Here is the local builds - https://github.com/Expensify/App/pull/65001#issuecomment-3302622285

**Comment by pasyukevich** [comment]
Updates: 

- QA list added 
- Videos uploaded 
- Adhoc builds are ready for testing - https://github.com/Expensify/App/pull/65001#issuecomment-3302622285

**Comment by IuliiaHerets** [comment]
@pasyukevich Issue is not reproducible with spec build

https://github.com/user-attachments/assets/1f09637e-f899-46bc-af4f-1564db6f62d0

**Comment by johncschuster** [comment]
@isabelastisser I'm currently focused on the Churn Project ([WN](https://expensify.slack.com/archives/CC7NECV4L/p1756299192400959)) and recently paused my membership on the BZ team (@laurenreidexpensify posted about this, [here](https://expensify.slack.com/archives/C041A90J23Z/p1757958561669749)) so I can maintain focus on that project. Can you take this the rest of the way? It should just be handling payment when the time is right.

**Comment by pasyukevich** [comment]
updates: 

 - All comments on PR addressed 
 - Waiting for  any issues/test ending from the QA with adhoc builds 

Tomorrow will be OOO, be back on Monday

**Comment by pasyukevich** [comment]
@johncschuster @isabelastisser

Just to clarify:  
   How will I know when this round of QA for ad-hoc builds is finished? 
   PR is ready to be open - Should I wait for QA before opening it?

**Comment by lakchote** [comment]
> [@johncschuster](https://github.com/johncschuster) [@isabelastisser](https://github.com/isabelastisser)
> 
> Just to clarify: How will I know when this round of QA for ad-hoc builds is finished? PR is ready to be open - Should I wait for QA before opening it?

@pasyukevich unfortunately, QA team wasn't informed that they needed to test your ad-hoc build.

I've just asked them here: https://expensify.slack.com/archives/C9YU7BX5M/p1758543085931009

They should provide an update on the related PR with the ad-hoc build: https://github.com/Expensify/App/pull/65001#issuecomment-3302622285

You can still open the PR by putting the `[HOLD]` label in its title (while you wait for QA results).

**Comment by pasyukevich** [comment]
PR is open - https://github.com/Expensify/App/pull/65001

**Comment by pasyukevich** [comment]
Updates: 

  - Got clarification for my questions here -  https://expensify.slack.com/archives/C07NMDKEFMH/p1758801808422989?thread_ts=1758645326.423979&cid=C07NMDKEFMH
  - Working on fixes : 
      https://github.com/Expensify/App/issues/71012 
      https://github.com/Expensify/App/issues/71014

**Comment by pasyukevich** [comment]
@nkuoch could you take a look at this?  

https://github.com/Expensify/App/issues/53441#issuecomment-3329046718

**Comment by pasyukevich** [comment]
Sure, thanks 

Let's clarify

**Comment by pasyukevich** [comment]
Asked here - https://expensify.com/archives/C07NMDKEFMH/p1759144468613929?thread_ts=1758645326.423979&cid=C07NMDKEFMH

**Comment by pasyukevich** [comment]
>   [#64876](https://github.com/Expensify/App/issues/64876) - notifications always show poisitive
> 
>   [#64896](https://github.com/Expensify/App/issues/64896) - delete message always coming positive from BE 
> 
>   `requestID: "9842885539d8026e-LAX"`
> 
>  <img width="814" height="435" alt="Image" src="https://github.com/user-attachments/assets/29186ed2-54c9-4918-bacd-4f10c1c39c09" />
> 
> In this case, the deleted amount was negative 

@nkuoch, Are there any updates on the other issues that might be API-related?

**Comment by nkuoch** [comment]
> https://github.com/Expensify/App/issues/64876 - notifications always show positive

Can you please retest and send me a screenshot of what you see? To me, backend is correctly returning negative values (but formatted with parentheses, no negative sign). So `-200` would display as `($200)` and it's expected

**Comment by pasyukevich** [comment]
Updates for today: 
   - Fixed as it was suggested- https://github.com/Expensify/App/issues/64896
   - Applying this change - https://expensify.com/archives/C07NMDKEFMH/p1759260299226359?thread_ts=1758645326.423979&cid=C07NMDKEFMH
   - Retesting notifications

Reposting the full list of actual issues (from here - https://github.com/Expensify/App/pull/65001#issuecomment-3324111461): 

Logged issues: 

[#71008](https://github.com/Expensify/App/issues/71008) - ✅ fixed  
[#71012](https://github.com/Expensify/App/issues/71012) - ✅ fixed 
[#71014](https://github.com/Expensify/App/issues/71014) - ✅ fixed 
[#71017](https://github.com/Expensify/App/issues/71017) -  ✅ fixed     
[#71042](https://github.com/Expensify/App/issues/71042) - (same as [#64896](https://github.com/Expensify/App/issues/64896))

KIs that failed the verification: 

[#64964](https://github.com/Expensify/App/issues/64964) -  ✅ fixed  
[#64868](https://github.com/Expensify/App/issues/64868) -  *recieved answer - https://expensify.com/archives/C07NMDKEFMH/p1759260299226359?thread_ts=1758645326.423979&cid=C07NMDKEFMH, applying* 
[#64876](https://github.com/Expensify/App/issues/64876) - (possibly BE related) - retesting  
[#64896](https://github.com/Expensify/App/issues/64896) - (possibly BE related) ✅ fixed  on FE side

**Comment by pasyukevich** [comment]
Updates for today: 
   - Fixed broken flow for transactions- https://github.com/Expensify/App/issues/64868 
   - Retesting issues after all fixes - do not see a blocker here for review
   - Asked for rereview - https://github.com/Expensify/App/pull/65001#issuecomment-3367202616

Reposting the full list of actual issues (from here - https://github.com/Expensify/App/pull/65001#issuecomment-3324111461): 

Logged issues: 

[#71008](https://github.com/Expensify/App/issues/71008) - ✅ fixed  
[#71012](https://github.com/Expensify/App/issues/71012) - ✅ fixed 
[#71014](https://github.com/Expensify/App/issues/71014) - ✅ fixed 
[#71017](https://github.com/Expensify/App/issues/71017) -  ✅ fixed     
[#71042](https://github.com/Expensify/App/issues/71042) - (same as [#64896](https://github.com/Expensify/App/issues/64896))

KIs that failed the verification: 

[#64964](https://github.com/Expensify/App/issues/64964) -  ✅ fixed  
[#64868](https://github.com/Expensify/App/issues/64868) -  ✅ fixed  on FE side - one small clarification, do not see a blocker  
[#64876](https://github.com/Expensify/App/issues/64876) - (possibly BE related) - was not able to reproduce on local build
[#64896](https://github.com/Expensify/App/issues/64896) - (possibly BE related) ✅ fixed  on FE side

**Comment by pasyukevich** [comment]
Updates for today: 

- Updated logic for report information in LHN 
-  Verified issues 
- Removed hold tag from PR 
- Fixed tests 

Ready to be reviewed

**Comment by ZhenjaHorbach** [comment]
Oh 
It was a long issue 
I already forgot that I started it on Upwork
And now I'm getting a payment on ND 😅

**Comment by isabelastisser** [comment]
@pac-guerreiro, what's your Upwork profile?

**Comment by pac-guerreiro** [comment]
@isabelastisser I'm not a contributor, there's no need to pay me 😉

**Comment by ZhenjaHorbach** [comment]
> There was [regression](https://github.com/Expensify/App/issues/72140) which was not handled by original C+ [@ZhenjaHorbach](https://github.com/ZhenjaHorbach). They were aware of that regression and didn't try to fix by themselves.

Just interesting 
Why are you so concerned about this issue? 😅
This issue was demoted and was open for external contributors(https://github.com/Expensify/App/issues/72140#issuecomment-3387025236)
And if this were a critical issue, the internal engineer could always assign the original developers

**Comment by isabelastisser** [comment]
`UPDATED` Payment summary:

@pasyukevich does not require payment (Contractor)
@waterim does not require payment (Contractor)
@pac-guerreiro does not require payment (Contractor)
@ZhenjaHorbach pending in NewDot for C+ review
